png-resizer/app/controllers/Resizer.scala

```diff
@@ -33,67 +33,87 @@ object Resizer extends Controller with Logging {
 
   case class PngResponse(status: Int, contentType: String, cacheHeaders: CacheHeaderList, body: Enumerator[Array[Byte]])
   type CacheHeaderList = Seq[(String, String)]
 
-  def getUpstreamResponse(uri: String, headers: Headers): FutureEither[PngResponse] = eitherTRight {
-    val unrolledHeaders = headers.getHeaders(Seq("Cache-Control", "If-Modified-Since", "If-None-Match"))
+  def getUpstreamResponse(uri: String, requestHeaders: Headers): Future[PngResponse] = {
+    val unrolledHeaders = requestHeaders.getHeaders(Seq("Cache-Control", "If-Modified-Since", "If-None-Match"))
     WS.url(uri).withHeaders(unrolledHeaders: _*).getStream().map {
       case (headers, enumerator) =>
         PngResponse(headers.status, headers.contentType, headers.getHeaders(Seq("Cache-Control", "Expires", "Last-Modified", "Etag")), enumerator)
     }
   }
 
-  def redirectScaleUpAttempts(originalWidth: Int, width: Int, fallbackUri: String) = eitherT (Future.successful {
+  def redirectScaleUpAttempts(originalWidth: Int, width: Int, fallbackUri: String) = {
     if (originalWidth <= width) {
-      logger.info(s"won't resize image to be bigger - $originalWidth to $width - redirecting to original")
+      log.info(s"won't resize image to be bigger - $originalWidth to $width - redirecting to original")
       -\/(Cached(1.day)(TemporaryRedirect(fallbackUri)))
     } else {
       \/-()
     }
-  })
+  }
 
   def resize(backend: String, path: String, widthString: String, qualityString: String) =
     Action.async { request =>
 
       (Backends.uri(backend, path), widthString, qualityString) match {
         case (Some(uri), IntString(width), IntString(quality)) =>
-          LoadLimit(uri) {
 
-            // Left here is the http result, right is the data that still needs computation
-            (for {
-              response <- Time(getUpstreamResponse(uri, request.headers), "download image", PngResizerMetrics.downloadTime)
-              cacheableContent <- getPngBytesToResize(uri, response)
-              CacheableContent(cacheHeaders, bytesPreResize) = cacheableContent
-              originalWidth <- eitherTRight(Im4Java.getWidth(bytesPreResize))
-              _ <- redirectScaleUpAttempts(originalWidth, width, uri)
-              resized <- Time(eitherTRight(Im4Java.resizeBufferedImage(width)(bytesPreResize)), "resize image", PngResizerMetrics.resizeTime)
-              quanted <- Time(eitherTRight(PngQuant(resized, quality)), "quantize image", PngResizerMetrics.quantizeTime)
-              result = Ok(quanted).as("image/png").withHeaders(cacheHeaders: _*)
-            } yield (result)).run.map(_.fold(identity,identity))
+          // Left here is the http result, right is the data that still needs computation
+          val resultEitherT = for {
+            response <- EitherT.right(Time(getUpstreamResponse(uri, request.headers), "download image", PngResizerMetrics.downloadTime))
+            cacheableContent <- eitherT(getPngBytesToResize(uri, response))
```

Comment:
Maybe it would be nicer to use `EitherT.apply` through just `EitherT` here. Then you can be definitely sure it's probably just wrapping it in the transformer. Otherwise I had to look it up and see that it was a method on `EitherT` that did the same thing :smile: